catalog/plugin/catalog-plugin-metacardbackup-s3storage/src/main/java/org/codice/ddf/catalog/plugin/metacard/backup/storage/s3storage/MetacardS3StorageRoute.java

```diff
@@ -0,0 +1,244 @@
+    public static final String S3_BUCKET_PROP = "s3Bucket";
+
+    public static final String S3_CANNED_ACL_NAME_PROP = "s3CannedAclName";
+
+    public static final String AWS_S3_ACCESS_KEY_PROP = "accessKey";
+
+    public static final String AWS_S3_SECRET_KEY_PROP = "secretKey";
+
+    public static final String AWS_S3_CLIENT_PROP = "amazonS3Client";
+
+    public static final String AWS_S3_ENDPOINT_PROP = "amazonS3Endpoint";
+
+    public static final String AWS_S3_DELETE_AFTER_WRITE_PROP = "deleteAfterWrite";
+
+    protected String objectTemplate;
+
+    protected String s3Bucket;
+
+    protected String s3AccessKey = "";
+
+    protected String s3SecretKey = "";
+
+    protected String s3Endpoint;
+
+    protected String s3CannedAclName;
+
+    private final org.apache.camel.impl.SimpleRegistry registry;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MetacardS3StorageRoute.class);
+
+    public MetacardS3StorageRoute(CamelContext camelContext) {
+        super(camelContext);
+        registry = new org.apache.camel.impl.SimpleRegistry();
+        CompositeRegistry compositeRegistry = new CompositeRegistry();
+        compositeRegistry.addRegistry(camelContext.getRegistry());
+        compositeRegistry.addRegistry(registry);
+        ((DefaultCamelContext) camelContext).setRegistry(compositeRegistry);
+    }
+
+    public String getObjectTemplate() {
+        return objectTemplate;
+    }
+
+    public void setObjectTemplate(String objectTemplate) {
+        this.objectTemplate = StringUtils.removeStart(objectTemplate, "/");
+    }
+
+    public String getS3Bucket() {
+        return s3Bucket;
+    }
+
+    public void setS3Bucket(String s3Bucket) {
+        this.s3Bucket = s3Bucket;
+    }
+
+    public String getS3AccessKey() {
+        return s3AccessKey;
+    }
+
+    public void setS3AccessKey(String s3AccessKey) {
+        this.s3AccessKey = s3AccessKey;
+    }
+
+    public String getS3SecretKey() {
+        return s3SecretKey;
+    }
+
+    public void setS3SecretKey(String s3SecretKey) {
+        this.s3SecretKey = s3SecretKey;
+    }
+
+    public String getS3Endpoint() {
+        return s3Endpoint;
+    }
+
+    public void setS3Endpoint(String s3Endpoint) {
+        this.s3Endpoint = s3Endpoint;
+    }
+
+    public String getS3CannedAclName() {
+        return s3CannedAclName;
+    }
+
+    public void setS3CannedAclName(String s3CannedAclName) {
+        this.s3CannedAclName = s3CannedAclName;
+    }
+
+    @Override
+    public void configure() throws Exception {
+        StringBuilder options = new StringBuilder();
+
+        if (StringUtils.isNotBlank(s3AccessKey)) {
+            addQuotedOption(options, AWS_S3_ACCESS_KEY_PROP, s3AccessKey);
+        }
+        if (StringUtils.isNotBlank(s3SecretKey)) {
+            addQuotedOption(options, AWS_S3_SECRET_KEY_PROP, "RAW(" + s3SecretKey + ")");
+        }
+
+        if (options.length() == 0) {
+            AmazonS3Client s3Client = new AmazonS3Client();
```

Comment:
It appears we will need to use this as Camel is trying to modify the object. The AmazonS3 constructed from the builder is immutable which causes problems.